WooCommerce/Classes/ViewRelated/ReusableViews/SwiftUI Components/JustInTimeMessageModal.swift

```diff
@@ -0,0 +1,114 @@
+import SwiftUI
+
+struct JustInTimeMessageModal: View {
+    let viewModel: JustInTimeMessageViewModel
+    @Binding var isPresented: Bool
+
+    var body: some View {
+        VStack(spacing: Layout.spacing) {
+            if let imageUrl = viewModel.imageUrl {
+                AdaptiveAsyncImage(lightUrl: imageUrl, darkUrl: viewModel.imageDarkUrl, scale: 3) { imagePhase in
+                    switch imagePhase {
+                    case .failure:
+                        Image(uiImage: viewModel.image)
+                            .accessibilityHidden(true)
+                    case .success(let image):
+                        image.resizable()
+                            .scaledToFit()
+                            .accessibilityHidden(true)
+                    case .empty:
+                        AnimatedPlaceholder()
+                    @unknown default:
+                        EmptyView()
```

Comment:
Should we add a log, or track this case somehow? Otherwise we might fail to catch on time new `AdaptiveAsyncImage` cases when/if are introduced

Reply:
No, because:

1. we can't do that in the middle of a `View.body`
2. we don't need to, because `@unknown default` means [the compiler will still produce a warning for any unhandled cases.](https://www.avanderlee.com/swift/unknown-default-enums-in-swift/#the-difference-between-default-and-unknown-default) So if another case gets added in future, we'll know about it when we build the app.